src/SSHLibrary/abstractclient.py

```diff
@@ -1085,7 +1086,8 @@ def put_file(self, sources, destination, mode, newline, path_separator='/'):
             contain the local path as the first value and the remote target
             path as the second.
         """
-        mode = int(mode, 8)
+        if isinstance(mode,int):
+            mode = int(mode, 8)
```

Comment:
As far as I can see, this breaks the old functionality. `mode` has always been string here, never an `int`. So now this if condition is never true with existing test cases, and actually will not work even if `mode` is changed to integer type.

We can tell `mode` never could have been `int` type even before, because `int(mode, 8)` for `int` type `mode` argument raises exception.
```
>>> mode = 0o0755
>>> int(mode, 8)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() can't convert non-string with explicit base
>>>
```

Reply:
I think something else is happening here. Because `mode = int(mode,8)` is not new. It has been there before, hence should always have raised the same exception for value `0o0755`

Reply:
So far it has always been a string, not int. Like
```
>>> mode = '0o0755'
>>> int(mode, 8)
493
>>>
```

Reply:
right. i confused the commits.